File: backend/domains/user/entity.go
package user

import (
	"time"

	"github.com/google/uuid"
)

type User struct {
	UserId    uuid.UUID
	UserName  string
	Email     Email
	CreatedAt time.Time
	UpdatedAt time.Time
}

func NewUser(userName string, email Email) User {
	userId := uuid.New()
	createdAt := time.Now()
	updatedAt := time.Now()

	return User{
		UserId:    userId,
		UserName:  userName,
		Email:     email,
		CreatedAt: createdAt,
		UpdatedAt: updatedAt,
	}
}

File: frontend/src/domains/user/entity.ts
import { z } from "zod";

export type User = {
  userId: string;
  userName: string;
  email: string;
  createdAt: Date;
  updatedAt: Date;
};

export const userSchema: z.ZodType<User> = z.object({
  userId: z.string().uuid(),
  userName: z.string(),
  email: z.string().email(),
  createdAt: z.date(),
  updatedAt: z.date(),
});
